test/tracing/mzcompose.py

```diff
@@ -0,0 +1,150 @@
+        "0:6878/api/stderr/config",
+        "-d",
+        '{"targets": "debug"}',
+    )
+    value = json.loads(
+        c.exec("materialized", "curl", "0:6878/api/tracing", capture=True).stdout
+    )
+    assert value["filter_fast_path"] == "debug"
+
+    # update the otel config
+    c.exec(
+        "materialized",
+        "curl",
+        "-X",
+        "PUT",
+        "-H",
+        "Content-Type: application/json",
+        "0:6878/api/opentelemetry/config",
+        "-d",
+        '{"targets": "trace"}',
+    )
+    value = json.loads(
+        c.exec("materialized", "curl", "0:6878/api/tracing", capture=True).stdout
+    )
+    assert value["filter_fast_path"] == "trace"
+
+    # revert the otel config and make sure we go back
+    c.exec(
+        "materialized",
+        "curl",
+        "-X",
+        "PUT",
+        "-H",
+        "Content-Type: application/json",
+        "0:6878/api/opentelemetry/config",
+        "-d",
+        '{"targets": "off"}',
+    )
+    value = json.loads(
+        c.exec("materialized", "curl", "0:6878/api/tracing", capture=True).stdout
+    )
+    assert value["filter_fast_path"] == "debug"
+
+    # make sure we can go allll the way back
+    c.exec(
+        "materialized",
+        "curl",
+        "-X",
+        "PUT",
+        "-H",
+        "Content-Type: application/json",
+        "0:6878/api/stderr/config",
+        "-d",
+        '{"targets": "info"}',
+    )
+    value = json.loads(
+        c.exec("materialized", "curl", "0:6878/api/tracing", capture=True).stdout
+    )
+    assert value["filter_fast_path"] == "info"
+
+
+def workflow_without_otel(c: Composition) -> None:
+    with c.override(Materialized()):
+        c.start_and_wait_for_tcp(services=["materialized"])
+
+        # Start with fastpath
+        value = json.loads(
+            c.exec("materialized", "curl", "0:6878/api/tracing", capture=True).stdout
+        )
+        assert value["filter_fast_path"] == "info"
+
+        # update the stderr config
+        c.exec(
+            "materialized",
+            "curl",
+            "-X",
+            "PUT",
+            "-H",
+            "Content-Type: application/json",
+            "0:6878/api/stderr/config",
+            "-d",
+            '{"targets": "debug"}',
+        )
+        value = json.loads(
+            c.exec("materialized", "curl", "0:6878/api/tracing", capture=True).stdout
+        )
+        assert value["filter_fast_path"] == "debug"
+        # make sure we can go allll the way back
+        c.exec(
+            "materialized",
+            "curl",
+            "-X",
+            "PUT",
+            "-H",
+            "Content-Type: application/json",
+            "0:6878/api/stderr/config",
+            "-d",
+            '{"targets": "info"}',
+        )
+        value = json.loads(
+            c.exec("materialized", "curl", "0:6878/api/tracing", capture=True).stdout
+        )
+        assert value["filter_fast_path"] == "info"
```

Comment:
Can you pivot this to pull out the internal port via `c.port` and then hit that port with the `requests` Python library? It'll be much easier to read than the `curl` invocations!